plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/JavaPackageValidator.java

```diff
@@ -13,27 +16,26 @@ public static boolean validate(String packageName) {
     } else if (packageName.endsWith(".")) {
       // todo or allow this and strip the period
       return false;
+    } else if (containsWhitespace(packageName)) {
+      // note very weird condition because validatePackageName allows internal white space
+      return false;
     } else {
-      String[] parts = packageName.split("\\.");
-      for (int i = 0; i < parts.length; i++) {
-        if (!isValidJavaName(parts[i])) {
-          return false;
-        }
-      }
+      return JavaConventions.validatePackageName(packageName, "1.4", "1.4").isOK();
```

Comment:
I think we should use whatever version we're generating in case there are other reserved words that come into play.  And we should the JavaCore constants ``JavaCore#VERSION_1_4``?